pkg/config/cmd/cmd.go

```diff
@@ -57,47 +108,112 @@ func HaltOnError(fn AfterFunc) AfterFunc {
 	}
 }
 
-func encodeAndCreate(info *resource.Info, namespace string, obj runtime.Object) (runtime.Object, error) {
+// Create is the default create operation for a generic resource.
+func Create(info *resource.Info, namespace string, obj runtime.Object) (runtime.Object, error) {
 	return resource.NewHelper(info.Client, info.Mapping).Create(namespace, false, obj)
 }
 
-// Create attempts to create each item generically, gathering all errors in the
-// event a failure occurs. The contents of list will be updated to include the
-// version from the server.
-func (b *Bulk) Create(list *kapi.List, namespace string) []error {
-	resourceMapper := &resource.Mapper{ObjectTyper: b.Typer, RESTMapper: b.Mapper, ClientMapper: resource.ClientMapperFunc(b.RESTClientFactory)}
-	after := b.After
-	if after == nil {
-		after = func(*resource.Info, error) bool { return false }
+func NoOp(info *resource.Info, namespace string, obj runtime.Object) (runtime.Object, error) {
+	return info.Object, nil
+}
+
+func labelSuffix(set map[string]string) string {
+	if len(set) == 0 {
+		return ""
+	}
+	return fmt.Sprintf(" with label %s", labels.SelectorFromSet(set).String())
+}
+
+func CreateMessage(labels map[string]string) string {
+	return fmt.Sprintf("Creating resources%s", labelSuffix(labels))
+}
+
+type BulkAction struct {
+	// required setup
+	Bulk        Bulk
+	Out, ErrOut io.Writer
+
+	// flags
+	Output      string
+	DryRun      bool
+	StopOnError bool
+
+	// output modifiers
+	Action string
+}
+
+// BindForAction sets flags on this action for when setting -o should only change how the operation results are displayed.
+// Passing -o is changing the default output format.
+func (b *BulkAction) BindForAction(flags *pflag.FlagSet) {
+	flags.StringVarP(&b.Output, "output", "o", "", "Output mode. Use \"-o name\" for shorter output (resource/name).")
+	flags.BoolVar(&b.DryRun, "dry-run", false, "If true, show the result of the operation without performing it.")
+}
+
+// BindForOutput sets flags on this action for when setting -o will not execute the action (the point of the action is
+// primarily to generate the output). Passing -o is asking for output, not execution.
+func (b *BulkAction) BindForOutput(flags *pflag.FlagSet) {
+	flags.StringVarP(&b.Output, "output", "o", "", "Output results as yaml or json instead of executing, or use name for succint output (resource/name).")
+	flags.BoolVar(&b.DryRun, "dry-run", false, "If true, show the result of the operation without performing it.")
+}
+
+// Compact sets the output to a minimal set
+func (b *BulkAction) Compact() {
+	b.Output = "compact"
```

Comment:
We have one use case today where we need to shorten output in a way
that differs from the -o - when you "install" a pod in new app, we
want to suppress everything but errors.  I think maybe I would do
something similar upstream or via a -q flag